src/domain/core/src/entities/working_datasets_map.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use opendatafabric::{DatasetHandle, DatasetID};
+
+use crate::Dataset;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Default)]
+pub struct WorkingDatasetsMap {
```

Comment:
The name stumbled me a bit when I saw it in function parameters. Although it consists of maps on the inside for fast lookups, the overall data structure seems more similar to a `Set<ResolvedDataset>` to me. If we were to implement mut methods on it, it would certainly behave like a set having `fn insert(&mut self, ds: ResolvedDataset)`.

Should we name it `ResolvedDatasetSet`?

Reply:
I like the idea of storing `ResolvedDataset` there. A bit tricker with handles iteration, but I managed. So now we have `ResolvedDatasetsMap`.